src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -0,0 +1,355 @@
+        this.httpClient = Objects.requireNonNull(httpClient);
+    }
+
+    @Override
+    public <T> Future<Void> processEvent(T event) {
+        final AuctionContext auctionContext;
+        final BidResponse bidResponse;
+
+        if (event instanceof AmpEvent ampEvent) {
+            auctionContext = ampEvent.getAuctionContext();
+            bidResponse = ampEvent.getBidResponse();
+        } else if (event instanceof AuctionEvent auctionEvent) {
+            auctionContext = auctionEvent.getAuctionContext();
+            bidResponse = auctionEvent.getBidResponse();
+        } else {
+            auctionContext = null;
+            bidResponse = null;
+        }
+
+        if (bidResponse == null || auctionContext == null) {
+            return Future.failedFuture("Bid response or auction context cannot be null");
+        }
+
+        final String greenbidsId = UUID.randomUUID().toString();
+        final String billingId = UUID.randomUUID().toString();
+
+        final Boolean isSampled = isSampled(greenbidsAnalyticsProperties.getGreenbidsSampling(), greenbidsId);
+
+        if (!isSampled) {
+            return Future.succeededFuture();
+        }
+
+        final CommonMessage commonMessage;
+        try {
+            commonMessage = createBidMessage(
+                    auctionContext,
+                    bidResponse,
+                    greenbidsId,
+                    billingId);
+        } catch (IllegalArgumentException e) {
+            return Future.failedFuture(e);
+        }
+
+        try {
+            final String commonMessageJson = jacksonMapper.encodeToString(commonMessage);
+
+            final MultiMap headers = MultiMap.caseInsensitiveMultiMap()
+                    .add(HttpUtil.ACCEPT_HEADER, HttpHeaderValues.APPLICATION_JSON)
+                    .add(HttpUtil.CONTENT_TYPE_HEADER, HttpHeaderValues.APPLICATION_JSON);
+
+            final Future<HttpClientResponse> responseFuture = httpClient.post(
+                    greenbidsAnalyticsProperties.getAnalyticsServer(),
+                    headers,
+                    commonMessageJson,
+                    greenbidsAnalyticsProperties.getTimeoutMs());
+
+            return responseFuture.compose(response -> {
+                final int responseStatusCode = response.getStatusCode();
+                if (responseStatusCode == 202 || responseStatusCode == 200) {
+                    return Future.succeededFuture();
+                } else {
+                    return Future.failedFuture(
+                            "Unexpected response status: " + response.getStatusCode());
+                }
+            });
+
+        } catch (EncodeException e) {
+            return Future.failedFuture("Failed to encode as JSON: " + e.getMessage());
+        }
+
+    }
+
+    private Boolean isSampled(double samplingRate, String greenbidsId) {
+        if (samplingRate < 0 || samplingRate > 1) {
+            logger.warn("Warning: Sampling rate must be between 0 and 1");
```

Comment:
as I can assume, you can just return false here, since it's not sampled in case of inappropriate rate?